src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java

```diff
@@ -72,6 +88,23 @@ public InternalDeliveryResult handleSmsRequest(final SmsRequest request) {
         return publishMessageEvent(message);
     }
 
+	public InternalDeliveryBatchResult handleSmsBatchRequest(final SmsBatchRequest request) {
+		var batchId = UUID.randomUUID().toString();
+
+		var deliveryResults = ofNullable(request.parties()).orElse(emptyList()).stream()
+			.filter(party -> isWhitelisted(SMS, party.mobileNumber()))
+			.map(party -> requestMapper.toSmsRequest(request, party))
+			.map(message -> messageMapper.toMessage(message, batchId))
+			.map(dbIntegration::saveMessage)
+			.map(message -> publishMessageEvent(message.withBatchId(batchId)))
```

Comment:
Is `batchId` already set with the call to `messageMapper.toMessage(message, batchId)` ?